experiment/kind-e2e.sh

```diff
@@ -71,9 +71,15 @@ build() {
     kind build node-image --type=bazel --kube-root="${PWD}"
 
     # make sure we have e2e requirements
-    #make all WHAT="cmd/kubectl test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo"
+    #make all WHAT="cmd/kubectl test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cluster/images/conformance/go-runner"
     bazel build //cmd/kubectl //test/e2e:e2e.test //vendor/github.com/onsi/ginkgo/ginkgo
 
+    # Temporary measure; don't fail on this. Only a stopgap to ensure the timing
+    # of this change and the corresponding one in k/k don't cause failures.
+    set +o pipefail
+    bazel build //cluster/images/conformance/go-runner || true
+    set -o pipefail
```

Comment:
Its my understanding that the build will use either the local directory (thats from the bazel build, right?) or the dockerized one as a fall back. Or do I have that confused and there is _another_ directory for the bazel output separate from `local` and `dockerized`?

My intention was that this would now:
 - use bazel to build the go-runner (and put it into the _output/local directory)
 - ignoring failures so that when it merges before the k/k change it wont fail other devs

Let me know if that isn't correct. On a mac and I've never gotten bazel to work properly for me so I'm sorry for not being able to test this exactly.

Reply:
> my understanding here is that ATM the runner binary is simply not built?

Correct, this repo just doesnt know about that new component so I need to build it. But if I add it here (without allowing for failure) then it will try and build it before the k/k repo has the files and it will still fail.

In an ideal world, the CI system would be able to realize that this PR in test-infra and the k/k PR are related and test them together.

> i don't think we need to tolerate the fails, because the e2e needs debugging.

Considering the statements above, I'm not sure I follow. Let me know if I'm misunderstanding, but I think I've clarified why the build fails and why we need to tolerate the pipefail for now. (A followup would remove that after both PRs go in)

Reply:
> Considering the statements above, I'm not sure I follow.

i guess it's OK to remove `|| true` later, but i still don't understand the need for it temporary.

> Its my understanding that the build will use either the local directory (thats from the bazel build, right?) or the dockerized one as a fall back.

i should place binaries in different dirs depending on the build tooling used.

if i do this under k/k:
```
bazel build //test/e2e:e2e.test
the binary will land in `bazel-bin/test/e2e/e2e.test`
```

> On a mac and I've never gotten bazel to work properly for me so I'm sorry for not being able to test this exactly.

quite frankly i don't see that many benefits of using bazel here.
especially given not everyone who wants to contribute is comfortable with it or being able test it locally.

i'd be happy to green light changes here to use non-bazel, but will defer to @BenTheElder and @dims on options first, as i think they touched these script last time.

Reply:
> bazel-bin

💯 I didnt know about that directory.
I can replace it with the make equivalent if others are OK with that and it will integrate with the dockerized/local output directories like the e2e conformance image is already using.

> but i still don't understand the need for it temporary.

I _want_ to build the go-runner permanently; the only temporary thing is building it here in a way that is permissive of failures (since this hasn't merged into k/k yet). If I dont do a temporary failure-permissive PR then either:
 - the k/k PR merges first but never passed this check. Maybe you're ok with that, in which case we don't really need a temporary step here.
 - This PR merges before the k/k one but then _everyones_ checks will fail this because the go-runner code doesn't exist in k/k yet.

I was trying to get this to merge first (in a way that wouldn't break others) so that my k/k change could actually pass this test.

Reply:
as @BenTheElder mentioned on slack this presubmit is not merge blocking.
thus technically, a change can be merged with errors and amended to pass later.

but yes having it pass on the PR is much better.